Code/Tools/FBuild/FBuildCore/Graph/DependencyListNode.cpp

```diff
@@ -0,0 +1,243 @@
+    REFLECT_ARRAY(  m_Patterns,                 "Patterns",                 MetaOptional() )
+    REFLECT_ARRAY(  m_PreBuildDependencyNames,  "PreBuildDependencies",     MetaOptional() + MetaFile() + MetaAllowNonFile() )
+REFLECT_END( DependencyListNode )
+
+// Anonymous DependencyListBuilder struct
+//------------------------------------------------------------------------------
+struct DependencyListBuilder {
+    enum {
+        iBuckets = 256,
+        mBuckets = (iBuckets - 1)
+    };
+
+    using bucket_t = Array< const Node * >;
+    bucket_t Visiteds[ iBuckets ];
+
+    const Array< AString > & Patterns;
+
+    explicit DependencyListBuilder( const Array< AString > & patterns )
+        : Patterns( patterns )
+    {
+    }
+
+    DependencyListBuilder( const DependencyListBuilder& ) = delete;
+    DependencyListBuilder& operator =( const DependencyListBuilder& ) = delete;
+
+    bool FindOrAddNode( const Node * node )
+    {
+        ASSERT( node );
+
+        bucket_t & bucket = Visiteds[ node->GetNameCRC() & mBuckets ];
+
+        if ( bucket.Find( node ) == nullptr )
+        {
+            bucket.Append( node );
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    void Collect( const Node * node )
+    {
+        if ( FindOrAddNode( node ) )
+        {
+            if ( node->GetType() == Node::ALIAS_NODE )
+            {
+                // resolve aliases to real nodes
+                AliasNode * aliasNode = node->CastTo< AliasNode >();
+                Collect( aliasNode->GetAliasedNodes() );
+            }
+
+            Collect( node->GetStaticDependencies() );
+            Collect( node->GetDynamicDependencies() );
+        }
+    }
+
+    void Collect( const Dependencies & deps )
+    {
+        for ( const Dependency & dep : deps )
+        {
+            if ( dep.IsWeak() == false )
+            {
+                Collect( dep.GetNode() );
+            }
+        }
+    }
+
+    bool IsMatch( const char * filename ) const
+    {
+        for ( const AString & pattern : Patterns )
+        {
+            if ( PathUtils::IsWildcardMatch( pattern.Get(), filename ) )
+            {
+                return true;
+            }
+        }
+        return Patterns.IsEmpty();
+    }
+
+    bool DumpTxtFile( const char* filename ) const
+    {
+        ASSERT( filename );
+
+        // Collect all file dependencies
+
+        Array< const AString * > dependencyList;
+
+        for ( const bucket_t & bucket : Visiteds )
+        {
+            for ( const Node * node : bucket )
+            {
+                if ( node->IsAFile() && IsMatch( node->GetName().Get() ) )
+                {
+                    dependencyList.Append( &node->GetName() );
+                }
+            }
+        }
+
+        // Sort collected filenames
+
+        dependencyList.Sort([]( const AString * a, const AString * b ) {
+            return ( a->Compare( *b ) < 0 );
+        });
```

Comment:
As a rule, FASTBuild code prefers "normal" functions over lambdas. This is for both style and practical reasons (as of 2019 the MSVC compiler has several off-by-one line number debug info bugs related to lambdas (as of VS2019 v14.25.28610))

Array implements SortDeref which avoid the need for this lambda, so that's easily resolved here.